extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/db/SQLBasicSecurityStorageConnector.java

```diff
@@ -0,0 +1,1123 @@
+          @Override
+          public Map<String, Object> inTransaction(Handle handle, TransactionStatus transactionStatus) throws Exception
+          {
+            return handle
+                .createQuery(
+                    StringUtils.format("SELECT * FROM roles where name = :roleName")
+                )
+                .bind("roleName", roleName)
+                .first();
+          }
+        }
+    );
+  }
+
+  @Override
+  public List<Map<String, Object>> getRolesForUser(String userName)
+  {
+    return getDBI().inTransaction(
+        new TransactionCallback<List<Map<String, Object>>>()
+        {
+          @Override
+          public List<Map<String, Object>> inTransaction(Handle handle, TransactionStatus transactionStatus)
+              throws Exception
+          {
+            List<Map<String, Object>> user_roles = handle
+                .createQuery(
+                    StringUtils.format(
+                        "SELECT roles.name\n"
+                        + "FROM roles\n"
+                        + "JOIN user_roles\n"
+                        + "    ON user_roles.role_name = roles.name\n"
+                        + "WHERE user_roles.user_name = :userName"
+                    )
+                )
+                .bind("userName", userName)
+                .list();
+            return user_roles;
+          }
+        }
+    );
+  }
+
+  @Override
+  public List<Map<String, Object>> getUsersWithRole(String roleName)
+  {
+    return getDBI().inTransaction(
+        new TransactionCallback<List<Map<String, Object>>>()
+        {
+          @Override
+          public List<Map<String, Object>> inTransaction(Handle handle, TransactionStatus transactionStatus)
+              throws Exception
+          {
+            List<Map<String, Object>> user_roles = handle
+                .createQuery(
+                    StringUtils.format(
+                        "SELECT users.name\n"
+                        + "FROM users\n"
+                        + "JOIN user_roles\n"
+                        + "    ON user_roles.user_name = users.name\n"
+                        + "WHERE user_roles.role_name = :roleName"
+                    )
+                )
+                .bind("roleName", roleName)
+                .list();
+            return user_roles;
+          }
+        }
+    );
+  }
+
+  private class PermissionsMapper implements ResultSetMapper<Map<String, Object>>
+  {
+    @Override
+    public Map<String, Object> map(int index, ResultSet resultSet, StatementContext context)
+        throws SQLException
+    {
+
+      int id = resultSet.getInt("id");
+      byte[] resourceJson = resultSet.getBytes("resource_json");
+      try {
+        final ResourceAction resourceAction = jsonMapper.readValue(resourceJson, ResourceAction.class);
+        return ImmutableMap.of(
+            "id", id,
+            "resourceAction", resourceAction
+        );
+      }
+      catch (IOException ioe) {
+        return null;
```

Comment:
Maybe better to return a map containing the exception?

Reply:
Made this propagate the exception since it should never happen under any normal conditions